src/components/toolSelfCheck/index.tsx

```diff
@@ -0,0 +1,873 @@
+                                                }
+                                                _hover={{
+                                                    bg: alphanumeric
+                                                        ? "black"
+                                                        : "white",
+                                                }}
+                                                color={
+                                                    alphanumeric
+                                                        ? "white"
+                                                        : "black"
+                                                }
+                                                background={
+                                                    alphanumeric
+                                                        ? "black"
+                                                        : "white"
+                                                }
+                                            >
+                                                Aa
+                                            </Button>
+                                            <Button
+                                                value={
+                                                    alphanumeric === true
+                                                        ? 1
+                                                        : 0
+                                                }
+                                                onClick={() =>
+                                                    onChangeAlphanumeric(
+                                                        questionId,
+                                                        false,
+                                                    )
+                                                }
+                                                _hover={{
+                                                    bg: alphanumeric
+                                                        ? "white"
+                                                        : "black",
+                                                }}
+                                                color={
+                                                    alphanumeric
+                                                        ? "black"
+                                                        : "white"
+                                                }
+                                                background={
+                                                    alphanumeric
+                                                        ? "white"
+                                                        : "black"
+                                                }
+                                            >
+                                                123
+                                            </Button>
```

Comment:
similar to the above comment- this block of code with the alphanumeric buttons is repeated I think 4 times, try moving it to the top level, if not, dw too much about it